### ISSUE TITLE
Lower-case `classname` prop for tags with hyphens

### ISSUE THREAD

**Issue by quentinvernot** [styled-components/styled-components#2480]
Really specific:
```js
const Element = styled('linear-gradient')``;
```
Will render:
```html
<linear-gradient classname="sc-gZMcBi giNnHm"></linear-gradient>
```

Notice the `classname` instead of `class`.

This only fails when calling `styled()` to render an element with a hyphen in the tag. Also fails when using the `as` prop. I tried it with all the hyphenated tags in `domElements.js` + `details-menu`, it fails for all of them. Putting the camelCase version of the tag (`linearGradient`) renders the wrong tag (`lineargradient` all attached) with the right prop. Calling the helper (`styled.linearGradient`) works great though.

Here is a sandbox with the issue: https://codesandbox.io/s/1z1k0jlny7

Am I missing something?

Edit: I'm using styled-components 4.2.0

**Comment by d7my11** [comment]
I had the same issue when I was working with AMP custom tags like `<amp-image className='test' />`.
I think it should be treated as Web Components in react and styled-component should pass `class` instead of `className`

> If you use React with Web Components (which is uncommon), use the class attribute instead.

**Comment by probablyup** [comment]
Does React itself handle hyphenated tag names differently? We generally follow whatever React does.

**Comment by quentinvernot** [comment]
> Am I missing something?

Oh, you have no idea, past me, you have no idea.



Sorry, finally had some time to come back to this, basically:

- The real tag is `<linearGradient />` (with a capital G for some reason), but `<lineargradient />` is perfectly fine too, the lib gets it right. There is no `<linear-gradient />`, my bad. Don't do CSS kids.
- `<details-menu />` is a [custom element](https://github.com/github/details-menu-element), styled-components doesn't know that and react just doesn't recognize it as a standard html5 tag, like @d7my11 said. Not the lib's fault as far I can tell.

`<linear-gradient />` is treated as a custom element, react doesn't like them, and everything is fine. Could be possible to fix custom elements in the lib, but that looks way beyond what I had in mind here.

In short: I messed up, sorry... Close?
